src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsReconciliationService.kt

```diff
@@ -17,15 +18,19 @@ class IncidentsReconciliationService(
   private val telemetryClient: TelemetryClient,
   private val incidentsService: IncidentsService,
   private val nomisIncidentsApiService: IncidentsNomisApiService,
+  @Value("\${reports.incidents.reconciliation.page-size}")
+  private val pageSize: Long = 20,
 ) {
   private companion object {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
   suspend fun generateReconciliationReport(agencies: List<IncidentAgencyId>): List<MismatchIncidents> =
-    withContext(Dispatchers.Unconfined) {
-      agencies.map { async { checkIncidentsMatch(it.agencyId) } }
-    }.awaitAll().filterNotNull()
+    agencies.chunked(pageSize.toInt()).flatMap { pagedAgencies ->
```

Comment:
The call to get all incident agency ids (which then use to call this method) could have been paged, but I didn't think it was needed so is now paged/chunked to avoid throttling the incidents api.  The list of agencies should be 120ish so this should still work ok.